pkg/mapper/dynamicfile/mapper.go

```diff
@@ -46,31 +47,35 @@ func (m *DynamicFileMapper) Map(identity *token.Identity) (*config.IdentityMappi
 	}
 
 	if roleMapping, err := m.RoleMapping(key); err == nil {
-		if err := m.match(identity, roleMapping.RoleARN, roleMapping.UserId); err != nil {
+		if err := m.match(canonicalARN, roleMapping.RoleARN); err != nil {
```

Comment:
So should we remove the match call to role since only user ARNs can change?

Reply:
thank you for the comments! yeah we can refactor it in future